internal/controller/object/object.go

```diff
@@ -420,39 +431,33 @@ func (c *external) setObserved(obj *v1alpha2.Object, observed *unstructured.Unst
 func (c *external) updateConditionFromObserved(obj *v1alpha2.Object, observed *unstructured.Unstructured) error {
 	switch obj.Spec.Readiness.Policy {
```

Comment:
nit: here we could declare ready and err before hand, just set them in the switch and move the code setting the conditions to after that.
```golang
func (c *external) updateConditionFromObserved(obj *v1alpha2.Object, observed *unstructured.Unstructured) error {
	var ready bool
	var err error
	switch obj.Spec.Readiness.Policy {
	case v1alpha2.ReadinessPolicyDeriveFromObject:
		ready = c.checkDeriveFromObject(observed)
	case v1alpha2.ReadinessPolicyAllTrue:
		ready = c.checkAllConditions(observed)
	case v1alpha2.ReadinessPolicyDeriveFromCelQuery:
		ready, err = c.checkDeriveFromCelQuery(obj, observed)
	case v1alpha2.ReadinessPolicySuccessfulCreate, "":
		// do nothing, will be handled by c.handleLastApplied method
		// "" should never happen, but just in case we will treat it as SuccessfulCreate for backward compatibility
	default:
		// should never happen
		return errors.Errorf("unknown readiness policy %q", obj.Spec.Readiness.Policy)
	}

	if err != nil {
		obj.SetConditions(xpv1.Unavailable().WithMessage(err.Error()))
		return nil
	}

	if !ready {
		obj.SetConditions(xpv1.Unavailable())
		return nil
	}

	obj.SetConditions(xpv1.Available())
	return nil
}
```

Reply:
Looks cleaner. I'm a fan.